### PR TITLE
scripts: west_commands: Remove deprecated west.log

### PR DESCRIPTION
**Pull request by pdgendt**
The global state `west.log` is deprecated, replace with `WestCommand` logging.

There are still more usages in tree, but those require more effort.

### CODE REVIEW

**Review by marc-hb — COMMENTED** [review]
I see nothing wrong but can you please "divide and conquer" this a bit? 

https://docs.zephyrproject.org/latest/contribute/contributor_expectations.html#defining-smaller-prs

Maybe split this across 2-4 different PRs? Also better from a _Continuous_ Integration perspective.

The sort of "stacked diffs" git-fu required for such a split should really not be a problem for a west maintainer :-P

**Comment by pdgendt** [comment]
> I see nothing wrong but can you please "divide and conquer" this a bit?

I've kept the trivial ones in this PR and will create separate ones for those with a larger "body".

**Comment by marc-hb** [comment]
Git conflict now (upvote https://github.com/orgs/community/discussions/4335 )

**Review by thedjnK — DISMISSED** [review]

**Comment by pdgendt** [comment]
Ping @marc-hb

**Comment by marc-hb** [comment]
> Ping @marc-hb

I try to help with `west` on my "spare time" and I've been spending a significant amount of time already. This API change LGTM but I know nothing about any of the scripts still in this PR. If there's no owner and reviewer for any of these scripts, not even for such a simple change, then I'm afraid these scripts have a maintenance problem bigger than this cleanup?

**Comment by pdgendt** [comment]
> > Ping @marc-hb
> 
> I try to help with `west` on my "spare time" and I've been spending a significant amount of time already. This API change LGTM but I know nothing about any of the scripts still in this PR. If there's no owner and reviewer for any of these scripts, not even for such a simple change, then I'm afraid these scripts have a maintenance problem bigger than this cleanup?

Reviewers are slim for these areas, I appreciate the efforts, but I do understand your argument as well.

File: scripts/west_commands/export.py [hunk 1]
```diff
@@ -44,25 +43,25 @@ def do_run(self, args, unknown_args):
...
-    elif pathobj.is_dir():
-        log.inf(f'- removing: {pathobj}')
-        rmtree(pathobj)
+    def remove_if_exists(self, pathobj):
+        if pathobj.is_file():
+            self.inf(f'- removing: {pathobj}')
+            pathobj.unlink()
+        elif pathobj.is_dir():
+            self.inf(f'- removing: {pathobj}')
+            rmtree(pathobj)
```
Comment (kartben) [inline scripts/west_commands/export.py:67]: this looks like dead code?

**Comment by mmahadevan108** [comment]
@pdgendt, for 4.0 we are only merging bug fixes and doc updates. Can this PR wait till after the release?

**Comment by pdgendt** [comment]
> @pdgendt, for 4.0 we are only merging bug fixes and doc updates. Can this PR wait till after the release?

Sure

File: scripts/west_commands/export.py (hunk 1 above)
Comment (pdgendt) [reply scripts/west_commands/export.py]: Removed

**Review by kartben — DISMISSED** [review]

**Review by kartben — APPROVED** [review]

**Review by thedjnK — APPROVED** [review]

**Pull request merged**